### PR TITLE
Bug #16264: When merging resources account for resource edges

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-graph/src/test/java/ai/grakn/graph/internal/PostProcessingTest.java [hunk 1]
```diff
@@ -148,7 +150,7 @@ public void whenMergingDuplicateResourcesWithRelations_EnsureSingleResourceRemai
 
         assertNotNull(foundR1);
         assertThat(foundR1.ownerInstances(), containsInAnyOrder(e1, e2, e3));
-        assertEquals(4, graknGraph.admin().getMetaRelationType().instances().size());
+        assertEquals(3, graknGraph.admin().getMetaRelationType().instances().size());
```
Comment (aelred) [inline grakn-graph/src/test/java/ai/grakn/graph/internal/PostProcessingTest.java]: Is this correct? The graph generated is the same as before.

File: grakn-graph/src/main/java/ai/grakn/graph/internal/ThingImpl.java [hunk 2]
```diff
@@ -201,7 +201,7 @@ public String getIndex(){
         Set<Role> roleSet = new HashSet<>(Arrays.asList(roles));
         Set<Relation> relations = new HashSet<>();
 
-        vertex().getEdgesOfType(Direction.OUT, Schema.EdgeLabel.RESOURCE).forEach(edge -> {
+        vertex().getEdgesOfType(Direction.BOTH, Schema.EdgeLabel.RESOURCE).forEach(edge -> {
```
Comment (aelred) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/ThingImpl.java]: D'oh. Surprised we both missed this!

File: grakn-graph/src/test/java/ai/grakn/graph/internal/PostProcessingTest.java (hunk 1 above)
Comment (fppt) [inline grakn-graph/src/test/java/ai/grakn/graph/internal/PostProcessingTest.java]: This is correct due to the "fakeness" of the test. The resource type is not linked properly for example

File: grakn-graph/src/main/java/ai/grakn/graph/internal/ThingImpl.java (hunk 2 above)
Comment (fppt) [inline grakn-graph/src/main/java/ai/grakn/graph/internal/ThingImpl.java]: I know :)

**Comment by fppt** [comment]
rtg

**Comment by fppt** [comment]
!rtg

**Comment by fppt** [comment]
Accidental merge of master into this branch. My gitfu is not strong enough to remove the accidental merge and rewrite the history. So I cherry picked the correct commits into a new branch: https://github.com/graknlabs/grakn/pull/1935

**Pull request closed without merging**